qiskit_ibm_runtime/ibm_runtime_service.py

```diff
@@ -249,6 +248,58 @@ def _resolve_credentials(
             account_credentials = credentials_list[0]
         return account_credentials, preferences
 
+    def _discover_remote_backends(
+        self, timeout: Optional[float] = None
+    ) -> Dict[str, "ibm_backend.IBMBackend"]:
+        """Return the remote backends available for this service instance.
+
+        Args:
+            timeout: Maximum number of seconds to wait for the discovery of
+                remote backends.
+
+        Returns:
+            A dict of the remote backend instances, keyed by backend name.
+        """
+        ret = OrderedDict()  # type: ignore[var-annotated]
+        backends_list = self._api_client.list_backends(timeout=timeout)
+        for backend_name in backends_list:
+            raw_config = self._api_client.backend_configuration(
+                backend_name=backend_name
+            )
+            # Make sure the raw_config is of proper type
+            if not isinstance(raw_config, dict):
+                logger.warning(
+                    "An error occurred when retrieving backend "
+                    "information. Some backends might not be available."
+                )
+                continue
+            try:
+                decode_backend_configuration(raw_config)
+                try:
+                    config = PulseBackendConfiguration.from_dict(raw_config)
+                except (KeyError, TypeError):
+                    config = QasmBackendConfiguration.from_dict(raw_config)
+                backend_cls = (
+                    ibm_backend.IBMSimulator
+                    if config.simulator
```

Comment:
The `GateConfig` object does not have a `simulator` attribute. Is that just a typing issue or a bug?
![image](https://user-images.githubusercontent.com/16404496/145551780-2a4dae46-0b3c-44bd-8a37-4229eb8e8ea1.png)

Reply:
I don't see this issue, functionality seems to work fine.

Reply:
Do you happen to know in which cases `config.simulator` is truthy? I don't see any assignments of `self.config = ` in `PulseBackendConfiguration` or `QasmBackendConfiguration` (or their parent classes).

Reply:
It’s truthy for simulators like ibmq_qasm_simulator and false for real backends. Not sure if that’s what you are asking?

Reply:
I was trying to understand where in the code `config.simulator` gets assigned (e.g. via `self.simulator`) in the backend classes or their parents.